logger/src/metrics.rs

```diff
@@ -44,39 +44,10 @@ pub trait Metric {
     fn count(&self) -> usize;
 }
 
-/// Representation of a metric that is expected  to be incremented from a single thread, so it
-/// can use simple loads and stores with no additional synchronization necessities.
-// Loads are currently Relaxed everywhere, because we don't do anything besides
-// logging the retrieved value (their outcome os not used to modify some memory location in a
-// potentially inconsistent manner). There's no way currently to make sure a SimpleMetric is only
-// incremented by a single thread, this has to be enforced via judicious use (although, every
-// non-vCPU related metric is associated with a particular thread, so it shouldn't be that easy
-// to misuse SimpleMetric fields).
-#[derive(Default)]
-pub struct SimpleMetric(AtomicUsize);
```

Comment:
In a perfect Firecracker we'd use something like `SimpleMetric` for all metrics that are only manipulated from a single thread. This would negligibly improve CPU cache performance. 

But that's such low priority that I'm totally fine with removing this unused code now.